contracts/Staking.sol

```diff
@@ -77,25 +69,27 @@ contract Staking is IERC20, IERC20Metadata {
         return totalAmount; 
     }
 
-    function balanceOf(address _user) public view returns(uint) {
-        UserInfo storage user = userInfo[_user];
-        uint acc = accamulatedRewardPerShare * user.amount - user.rewardDebt;
-        return user.accamulatedReward + acc + user.amount;
+    function rewardPerSecond() public view returns (uint) {
+        return delta() * aprBasisPoints * calcDecimals / 10000 / secondsInYear;
     }
 
-    function setAdmins(address[] memory admins) public onlyOwner {
-        for(uint i = 0; i < admins.length; i++) {
-            lpAdmins.push(admins[i]);
-        }
+    function delta() internal view returns (uint) {
+        return block.timestamp - lastRewardTimestamp;
     }
 
-    function removeAdmins(address[] memory admins) public onlyOwner {
-        for(uint i = 0; i < admins.length; i++) {
-            AddressArrayLib.removeItem(lpAdmins, admins[i]);
-        }
+    function calculateRewardAmount(uint amount) internal view returns (uint) {
+        return accumulatedRewardPerShare * amount / calcDecimals;
     }
 
-    function setApr(uint _aprBasisPoints) public onlyAdmin {
+    function balanceOf(address _user) public view returns(uint) {
+        UserInfo storage user = userInfo[_user];
+        uint updAccumulatedRewardPerShare = rewardPerSecond() + accumulatedRewardPerShare;
```

Comment:
Shouldn't we multiply rewardPerSecond() by the time diff here?

Reply:
Changed fn name to lastDeltaReward